app/src/main/java/com/gh4a/fragment/RepositoryFragment.java

```diff
@@ -385,15 +438,104 @@ private void loadPullRequestCount(boolean force) {
+                    OverviewRow issuesRow = mContentView.findViewById(R.id.issues_row);
+                    issuesRow.setText(getResources().getQuantityString(R.plurals.issue, issueCount, issueCount));
 
-                    TextView tvPullRequestsCountView = v.findViewById(R.id.tv_pull_requests_count);
-                    tvPullRequestsCountView.setText(String.valueOf(count));
+                    OverviewRow pullsRow = mContentView.findViewById(R.id.pulls_row);
+                    pullsRow.setText(getResources().getQuantityString(R.plurals.pull_request, count, count));
                 }, this::handleLoadFailure);
     }
+
+    private void toggleStarringState() {
+        StarringService service = ServiceFactory.get(StarringService.class, false);
+        Single<Response<Void>> responseSingle = mIsStarring
+                ? service.unstarRepository(mRepository.owner().login(), mRepository.name())
+                : service.starRepository(mRepository.owner().login(), mRepository.name());
+        responseSingle.map(ApiHelpers::mapToBooleanOrThrowOnFailure)
+                .compose(RxUtils::doInBackground)
+                .subscribe(result -> {
+                    if (mIsStarring != null) {
+                        mIsStarring = !mIsStarring;
+                        mRepository = mRepository.toBuilder()
+                                .stargazersCount(mRepository.stargazersCount() + (mIsStarring ? 1 : -1))
+                                .build();
+                        updateStargazerUi();
+                    }
+                }, error -> {
+                    handleActionFailure("Updating repo starring state failed", error);
+                    updateStargazerUi();
+                });
+
+    }
+
+    private void toggleWatchingState() {
+        WatchingService service = ServiceFactory.get(WatchingService.class, false);
+        final String repoOwner = mRepository.owner().login(), repoName = mRepository.name();
+        final Single<?> responseSingle;
+
+        if (mIsWatching) {
+            responseSingle = service.deleteRepositorySubscription(repoOwner, repoName)
+                    .map(ApiHelpers::throwOnFailure);
+        } else {
+            SubscriptionRequest request = SubscriptionRequest.builder()
+                    .subscribed(true)
+                    .build();
+            responseSingle = service.setRepositorySubscription(repoOwner, repoName, request)
+                    .map(ApiHelpers::throwOnFailure);
+        }
+
+        responseSingle.compose(RxUtils::doInBackground)
+                .subscribe(result -> {
+                    if (mIsWatching != null) {
+                        mIsWatching = !mIsWatching;
+                        mRepository = mRepository.toBuilder()
+                                .subscribersCount(mRepository.subscribersCount() + (mIsWatching ? 1 : -1))
+                                .build();
+                        updateStargazerUi();
```

Comment:
Just as a reminder this line and line 500 should still be `updateWatcherUi`.

Reply:
Done, thanks!